src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -40,20 +44,32 @@ public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
         /// <inheritdoc />
         public IDirectoryContents GetDirectoryContents(string subpath)
         {
-            if (!StartsWithBasePath(subpath, out var physicalPath))
+            var modifiedSub = NormalizePath(subpath);
+
+            if (StartsWithBasePath(modifiedSub, out var physicalPath))
             {
-                return NotFoundDirectoryContents.Singleton;
+                return InnerProvider.GetDirectoryContents(physicalPath.Value);
+            }
+            else if (string.Equals(subpath, string.Empty))
+            {
+                return new StaticWebAssetsDirectoryRoot(BasePath);
+            }
+            else if (BasePath.StartsWithSegments(modifiedSub, FilePathComparison, out PathString remaining))
+            {
+                return new StaticWebAssetsDirectoryRoot(remaining);
```

Comment:
```suggestion
                return new StaticWebAssetsDirectoryRoot(remaining.Value);
```

Reply:
Did you applied this?

Reply:
I responded to that in the second paragraph [here](https://github.com/aspnet/AspNetCore/pull/14803#discussion_r338269785). Is there a reason why that wouldn't be correct?

Reply:
Yes. If it has a space in it it will encode it as "%20" and that will make comparisons later fail.

Reply:
I don't understand how that can be the case when `StaticWebAssetsDirectoryRoot` [takes a `PathString`](https://github.com/aspnet/AspNetCore/pull/14803/files#diff-11cbff5b8d8860e80be8908027ec3c07R101) that it [uses the `Value` property](https://github.com/aspnet/AspNetCore/pull/14803/files#diff-11cbff5b8d8860e80be8908027ec3c07R104) off of. I could change `StaticWebAssetsDirectoryRoot` to take a `string`, but then we have to get the use of `PathString.Value` correct in 2 places (and any future place) instead of inside the constructor where we can get it right only once and leave a warning comment.

I created a [test of this scenario](https://github.com/aspnet/AspNetCore/pull/14803/commits/8825b1948709cc1b57b74c91d97de7b9c85edec7#diff-9bc08090a413386dd1a4a2fee1b7346dR92) to ensure that it works as expected.

Reply:
@javiercn are we good here or is there something I'm not seeing?

Reply:
No, you are right, it's hard to see from the code and I was being cautious because there's an implicit conversion between PathString and string, but as you mention it doesn't apply to this case.